lms/models/grouping.py

```diff
@@ -23,6 +23,10 @@ class Type(str, Enum):
     __table_args__ = (
         sa.UniqueConstraint("application_instance_id", "authority_provided_id"),
         sa.UniqueConstraint("lms_id", "application_instance_id", "parent_id", "type"),
+        sa.CheckConstraint(
```

Comment:
There's also https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#sqlalchemy.dialects.postgresql.ENUM

I would pick which ever plays nicer with alembic (as far as I remember alembic didn't detect changes in the type so the explicit constraint might be simpler)